### DISCUSSION TITLE
[Bug] OSX arm64 builds appear to not run

### DISCUSSION THREAD

**Discussion by apullin** [sn4k3/UVtools#795] [Q&A] [answered]
### System

```Text
UVtools v4.0.4 X64
Operative system: Darwin 23.1.0 Darwin Kernel Version 23.1.0: Mon Oct  9 21:28:45 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6020 X64
Processor: Apple M2 Max
Processor cores: 12
Memory RAM: 12.91 / 27.72 GB
Runtime: osx.14-x64
Framework: .NET 6.0.24
AvaloniaUI: 11.0.5
OpenCV: 4.8.0

Sreens, resolution, working area, usable area:
1: 1728 x 1117 @ 100% (Primary)
    WA: 1728 x 1023    UA: 1728 x 1023

Path:       /Applications/UVtools.app/Contents/MacOS/
Executable: /Applications/UVtools.app/Contents/MacOS/UVtools
Loaded file: stock.pm5s [Version: 518] [Class: PhotonWorkshopFile]
```

### Printer and Slicer

- Printer: 
- Slicer: 

### Description of the bug

The OSX arm64 builds appears to not run. I tested a variety of versions, as far back as 3.6.0 .

The only error output is a simple dialog with:
"The application “UVtools” can’t be opened."

This is AFTER permitting the app to run, via the normal quarantine allowance, "Open anyway".

NOTE: the System Information above is for the x86_64 version, which DOES run (through Apple's emulation, I assume).

### How to reproduce

1. Download and unpack [UVtools_osx-arm64_v4.0.4.zip](https://github.com/sn4k3/UVtools/releases/download/v4.0.4/UVtools_osx-arm64_v4.0.4.zip) onto an Apple Silicon mac
2. attempt to run it

**Comment by sn4k3** [comment, marked as answer]
You can't install by your own, macOS now block all unverified apps.
You need to install it via my script, it will bypass security.
https://github.com/sn4k3/UVtools#to-auto-install-on-macos

**Reply by apullin** [reply]
Ah, ok. Notably, I saw this error when running:
```
- Codesign app bundle
The following argument was not expected: --deep
```

The issue is that my anaconda python distribution installed its own version of `codesign`.
Deactivating that env, now `codesign` is the bundled OS tool at `/usr/bin/codesign`, and everything works fine!

**Comment by BenjaminVanRyseghem** [comment]
You can self-sign the app using

```
xattr -cr /Applications/UVtools.app && codesign --force --deep --sign - /Applications/UVtools.app 
```
